spring-kafka/src/test/java/org/springframework/kafka/support/serializer/RetryingDeserializerTests.java

```diff
@@ -54,6 +59,19 @@ void testRetry() {
 		rdes.close();
 	}
 
+	@SuppressWarnings("unchecked")
+	@Test
+	void retryingDeserializerWithRecoveryCallback() throws Exception {
+		RetryingDeserializer<String> rdes = new RetryingDeserializer<>((s, b) -> {
+			throw new RuntimeException();
+		}, new RetryTemplate());
+		RecoveryCallback<String> recoveryCallback = mock(RecoveryCallback.class);
+		rdes.setRecoveryCallback(recoveryCallback);
+		rdes.deserialize("my-topic", "my-data".getBytes());
+		verify(recoveryCallback, times(1)).recover(any(RetryContext.class));
+		rdes.close();
```

Comment:
There is nothing to close. See our delegate lambda.
IMHO the test should be as short as possible and be focused on the specific task.
If tests are too complicated and they are flaky, it is hard to find the reason and solution.
I know: this one is not, but better to have a good habit for less code - better support!

Reply:
ok, will cleanup, i was just following original test patterns.